databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -166,13 +169,17 @@ def _decorate_staleness(self,
         :return:
         """
         if self.ms_to_expire:
-            return statement.format(staleness_condition=textwrap.dedent(f"""\
-            (target.publisher_last_updated_epoch_ms < (timestamp() - ${MARKER_VAR_NAME})
-            OR NOT EXISTS(target.publisher_last_updated_epoch_ms))"""))
-
-        return statement.format(staleness_condition=textwrap.dedent(f"""\
-        (target.published_tag <> ${MARKER_VAR_NAME}
-        OR NOT EXISTS(target.published_tag))"""))
+            condition = f"""(target.publisher_last_updated_epoch_ms < (timestamp() - ${MARKER_VAR_NAME})"""
+            condition = (condition + ")"
+                         if self.retain_data_with_no_publisher_metadata
+                         else condition + "\nOR NOT EXISTS(target.publisher_last_updated_epoch_ms))")
```

Comment:
how about:

```
condition = f"""(target.publisher_last_updated_epoch_ms < (timestamp() - ${MARKER_VAR_NAME})"""
if not self.retain_data_with_no_publisher_metadata:
  null_check = "EXISTS(target.publisher_last_updated_epoch_ms)"
  condition = f"{condition} OR NOT {null_check}"
```

and likewise for the ds-only case. 

Slightly cleaner, maybe. wdyt.